### ISSUE TITLE
Operating System becomes extremely slow when parsing the default C++ includes

### ISSUE THREAD

**Issue by j-stephan** [microsoft/vscode-cpptools#36]
This is (partly) a duplicate of #26 and #24. On Linux the default include path is "/usr/include" which is fine for standard C programs. However, the C++ STL is missing by default (see #24) and has to be included in c_cpp_properties.json manually. Once this is done Visual Studio Code will start parsing the newly found includes (I assume those in /usr/include as well) which somehow makes the rest of the OS absurdely slow - even vim takes ages to respond. I assume this is due to high disk usage.

If I leave the system alone it will take about an hour until parsing is completed - resulting in a database file which is about 1GiB in size **per project**. This is where #26 kicks in: if I close and reopen Visual Studio Code the old database will be deleted and the whole process starts once again.

Not parsing the default includes is possible but those "include not found" warnings get annoying very fast.

**Comment by sridmad** [comment]
In 0.7.0 we added the ability to control the files that are processed and added to the database.

/usr/include - recursively add all files under the folder to the database
 /usr/include/\* - Do not recurse folders under /usr/include, just add the files in /usr/include to database

This should allow finer grained control of the files processed through judicious use of includepaths

We are also adding another switch to allow adding just the files referenced (directly / indirectly) by files under the root folder. This should be enabled in 0.8.0 version of the extension

**Comment by sridmad** [comment]
Both of the features should help reduce the number of files processed and the stress on the system. All files are processed in a single thread so most of the impact seems to be due to I/O, reading headers and writing to DB. If you continue to see perf issue after the above features are enabled, we can look for other mitigations.

**Comment by jgoshi** [comment]
Could you try this again with the latest release (VSCode 1.4, cpptools 0.8.1) and let us know if the situation has improved?

**Comment by j-stephan** [comment]
Yes, the performance issue seems to be fixed. I'm using the default values for "c_cpp_properties.json" and everything works like a charm.
